manim/mobject/three_d/three_dimensions.py

```diff
@@ -715,6 +722,15 @@ def get_direction(self) -> np.ndarray:
         """
         return self.direction
 
+    def _set_start_and_end_attributes(self, direction):
+        normalized_direction = direction * np.linalg.norm(direction)
+
+        start = self.base_circle.get_center()
+        end = start + normalized_direction * self.new_height
+        self.start_point = VMobject().set_points_as_corners([start] * 3)
+        self.end_point = VMobject().set_points_as_corners([end] * 3)
```

Comment:
Can the tripled reference to `start` and `end` cause any harm here? (Modifying any of the entries will modify all three of them in the same way.)

Reply:
I think that it must be the same point three times. Otherwise we will get a 3d curve wich would be visible, and we don't want this. My solution based on merge #3358 or #3718. I am not sure why we might have problem here, because we are calculating everything internally based on user start and end points. Maybe I missunderstood you question

Reply:
I believe he is talking about this behavior
```py
x = [1]
y = [x]*3
x.append(2)
print(y)
```
Which results in
```py
[[1, 2], [1, 2], [1, 2]]
```

Reply:
Exactly, this is what I was referring to!

Reply:
Okay, but should we care about it? They are local variables, as long as direction, get_center and new_height will return propper types we will not have a problem here. I could add type checking but I am not sure that we need it here. Why? Because if direction won't have a proper type cone and cilinder will through the error same for height, if get_center will have invalid type all manim will be broken. I could add it but IMO if the type of the variables used in this method other part of the code will through the error ealier.

Reply:
Let me rephrase my question: can it happen that just one of the three entries of `self.start_point` or `self.end_point` is modified while the other two should stay the same?

Reply:
We can't prevent user to modify these member variables. The user shouldn't touch these variables, except get_start and get_end methods. But in the existing code I think there are no possibility to break this.

Reply:
Sorry for being a bit slow! Actually, I also don't quite understand yet: why is `start_point` / `end_point` a `VMobject` with three corners, instead of just a `VectorizedPoint` with the given `start` and `end` coordinates?

Reply:
1. I didn’t know that VectorizedPoint exist, tommorow I could change it to VectorizedPoint 
2. I talked one or two weeks ago with Mr.Diver and I implemented his "advice"
3. I quickly checked the implementation of VectorizedPoint and it almost the same. But for some type of convention I think that it would be better to have a VectorizedPoint instead of existing implementation.

The decision is yours.

Reply:
I don't want to cause any worries or block this for no reason, I just find the construction here a bit odd. Can you explain why these mobjects here are paths that visit the same point three times? I'm just curious about why `[start]*3` is necessary.

Reply:
I'm sorry for not getting back to you sooner. I checked we could use VectorizedPoint, and it will work. I just thought we need at least three points to construct any manim object. I can't run the tests locally. There are some problems with the .toml file, but everything works correctly when I run the standard manim script.

Reply:
Hello!
I'd also want this to get merged soon.

Regarding the initial concern, `VMobject.set_points_as_corners()` creates a new NumPy array from scratch using the given points, so that `VMobject.points` no longer contains 3 references to the same point:

```py
>>> vmob = VMobject().set_points_as_corners([UP] * 3)
>>> vmob.points
array([[0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.]])
>>> vmob.points[0][2] = 5
>>> vmob.points
array([[0., 1., 5.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.],
       [0., 1., 0.]])
```
So it's OK, you don't have to worry about that :)

Reply:
So, will the implementation be changed to a `VectorizedPoint` or similar, or will it stay the same? 👀

Reply:
I will push today the VectorizedPoint version.

Reply:
Is there any update on this? It would be nice to have it merged before the v0.19.0 release :)

Reply:
I had problems with my laptop (and I was finishing my 4th physic semester)
I will push tuday the newest version of the code
Sorry

Reply:
i just switched to new PC, i will install manim tomorrow and then i will add changes -> push.|
For some reason the changes which i made, wasn't saved, i have no idea why

Reply:
I have seen the changes, thanks!